src/modules/wallet/sagas.ts

```diff
@@ -199,8 +203,22 @@ function* handleSwitchNetworkRequest(action: SwitchNetworkRequestAction) {
     )
   } else {
     try {
-      yield call(switchProviderChainId, provider, chainId)
-      yield put(switchNetworkSuccess(chainId))
+      const { timeout } = yield race({
+        switched: call(switchProviderChainId, provider, chainId),
+        timeout: delay(SWITCH_NETWORK_TIMEOUT) // 10 seconds timeout
+      })
+
+      if (timeout) {
+        yield put(
+          switchNetworkFailure(
+            chainId,
+            'Error switching network: Operation timed out'
+          )
+        )
```

Comment:
If you want, you can avoid repeating the failure action by throwing:
```suggestion
      if (timeout) {
        throw new Error('Error switching network: Operation timed out')
```

Reply:
great idea! updated!